src/JWT.ts

```diff
@@ -326,16 +343,94 @@ export async function createJWT(
     }
   }
   const fullPayload = { ...timestamps, ...payload, iss: issuer }
-  return createJWS(fullPayload, signer, header, { canonicalize })
+  return createJWS(fullPayload, signer, header, { canonicalize }) as Promise<string>
+}
+
+/**
+ *  Creates a multi-signature signed JWT given multiple issuers and their corresponding signers, and a payload for which the signature is
+ * over.
+ *
+ *  @example
+ *  const signer = ES256KSigner(process.env.PRIVATE_KEY)
+ *  createJWT({address: '5A8bRWU3F7j3REx3vkJ...', signer}, {key1: 'value', key2: ..., ... }).then(jwt => {
+ *      ...
+ *  })
+ *
+ *  @param    {Object}            payload               payload object
+ *  @param    {Object}            [options]             an unsigned credential object
+ *  @param    {boolean}           options.expiresIn     optional flag to denote the expiration time
+ *  @param    {boolean}           options.canonicalize  optional flag to canonicalize header and payload before signing
+ *  @param    {Object[]}          issuers               array of the issuers, their signers and algorithms
+ *  @param    {string}            issuers[].issuer      The DID of the issuer (signer) of JWT
+ *  @param    {Signer}            issuers[].signer      a `Signer` function, Please see `ES256KSigner` or `EdDSASigner`
+ *  @param    {String}            issuers[].alg         [DEPRECATED] The JWT signing algorithm to use. Supports:
+ *   [ES256K, ES256K-R, Ed25519, EdDSA], Defaults to: ES256K. Please use `header.alg` to specify the algorithm
+ *  @return   {Promise<Object, Error>}                  a promise which resolves with a signed JSON Web Token or
+ *   rejects with an error
+ */
+export async function createMultisignatureJWT(
+  payload: Partial<JWTPayload>,
+  { expiresIn, canonicalize }: Partial<JWTOptions>,
+  issuers: { issuer: string; signer: Signer; alg: string }[]
+): Promise<string> {
+  if (issuers.length === 0) throw new Error('invalid_argument: must provide one or more issuers')
+
+  let payloadResult: Partial<JWTPayload> = payload
+
+  let jwt = ''
+  for (let i = 0; i < issuers.length; i++) {
+    const issuer = issuers[i]
+
+    const header: Partial<JWTHeader> = {
+      typ: 'JWT',
+      alg: issuer.alg,
+    }
+
+    // Create nested JWT
+    // See Point 5 of https://www.rfc-editor.org/rfc/rfc7519#section-7.1
+    // After the first JWT is created (the first JWS), the next JWT is created by inputting the previous JWT as the payload
+    if (i !== 0) {
+      header.cty = 'JWT'
+    }
+
+    jwt = await createJWT(payloadResult, { ...issuer, canonicalize, expiresIn }, header)
+
+    payloadResult = { jwt }
+  }
+  return jwt
 }
 
-function verifyJWSDecoded(
-  { header, data, signature }: JWSDecoded,
+export function verifyJWSDecoded(
+  { header, payload, data, signature }: JWTDecoded,
```

Comment:
This should be reverted as it would change the expected input for `verifyJWS`, which is an exported method.

`verifyJWS*` methods shouldn't expect the payload to decode to anything, and just treat it as a blob of bytes.
This would also remove the need to cast the data types in weird ways (`jwsDecoded as unknown as JWTDecoded`)
The logic you added should be performed at a higher level. Perhaps in `verifyJWT`?

Reply:
I have now reverted `verifyJWSDecoded()` back to original function and created a new function `verifyJWTDecoded()` instead with the required changes.